modules/ROOT/pages/anypoint-mq/3.x/anypoint-mq-publish.adoc

```diff
@@ -0,0 +1,103 @@
+
+image::anypoint-mq/3.x/amq-3x-publish-operation.png[Anypoint MQ Publish Operation]
+
+[source,xml]
+----
+<anypoint-mq:publish destination="invoiceQueue" config-ref="Anypoint_MQ_config"/>
+----
+
+This operation produces the following output:
+
+* A copy of the body sent as payload.
++
+For example, if you are sending a "Hello world" message, the resulting payload is the "Hello world" String.
+* The message's `messageId` in the message attributes.
+
+image::anypoint-mq/3.x/amq-3x-publish-attributes.png[Anypoint MQ Publish Operation Output Attributes]
+
+
+== Configure User Properties
+
+You can define properties for outgoing messages, for example, to provide compatibility with other messaging systems or to communicate the content-type of a message.
+
+Use the `properties` parameter of the Publish operation to configure these properties directly in the message. Define the `properties` parameter as a key/value map of Strings using DataWeave:
+
+image::anypoint-mq/3.x/amq-3x-publish-properties.png[Anypoint MQ Publish With Properties]
+
+[source,xml,linenums]
+----
+<anypoint-mq:publish config-ref="Anypoint_MQ_Config"
+    destination="invoiceQueue">
+  <anypoint-mq:properties><![CDATA[#[output application/java ---
+      {
+          AUTH_TYPE: 'jwt',
+          AUTH_TOKEN: attributes.queryParams.token
+        }]]]>
+      <anypoint-mq:properties>
+</anypoint-mq:publish>
+----
+
+=== Transform the Message Body
+
+If the payload is not in the correct format, you can transform it to the correct format.
+
+To transform a payload, declare an inline DataWeave transformation in the `body` parameter of the Publish operation:
+
+image::anypoint-mq/3.x/amq-3x-publish-transformed.png[Anypoint MQ Publish With Transformation]
+
+[source,xml,linenums]
+----
+<anypoint-mq:publish config-ref="Anypoint_MQ_Config"
+   destination="invoiceQueue">
+      <anypoint-mq:body ><![CDATA[#[output application/json ---
+      {
+        customer: payload.id,
+        items: payload.invoice.items
+      }]]]>
+      </anypoint-mq:body>
+</anypoint-mq:publish>
+----
+
+In this case, the result of the operation is the _transformed_ value as JSON.
+
+== Advanced Configurations
+
+
+=== Set a Custom Message ID
+
+Instead of letting the Anypoint MQ service generate a unique `messageId` for the new published message, you can set a custom `messageId` for your outgoing message using the *Message id* field in the *Advanced* tab of the main screen:
+
+image::anypoint-mq/3.x/amq-3x-publish-advanced.png[Anypoint MQ Publish Advanced Tab]
+
+WARNING: The ID of an Anypoint MQ message must be unique. Make sure to choose a unique custom ID to avoid unwanted side effects of duplicated IDs. In FIFO queues, messages with duplicate IDs are overwritten.
+
+=== Propagate Content Type
+
+To specify whether the Content Type of the message body is propagated as a *property* of the Anypoint MQ message or not, use the `sendContentType` parameter.
```

Comment:
Why init. cap. Content Type? If it is a name, it needs a common noun object.

Reply:
ditto *property* : Supply a noun object, or use the common noun "property."
If this is intended to be bold for emphasis, unbold it.